Stack/Opc.Ua.Core/Stack/Tcp/TcpTransportListener.cs

```diff
@@ -801,6 +801,31 @@ private void OnAccept(object sender, SocketAsyncEventArgs e)
                     {
                         // TODO: .Count is flagged as hotpath, implement separate counter
                         int channelCount = channels.Count;
+
+                        // Remove oldest channel that does not have a session attached to it
+                        // before reaching m_maxChannelCount
+                        if (m_maxChannelCount > 0 && m_maxChannelCount == channelCount)
+                        {
+                            var snapshot = channels.Values.ToArray();
```

Comment:
Please check but I *think* the snapshot works only on the full thing.
snapshot = channels.ToArray();
etcetc

Reply:
Yes, from what I read you are correct.